### ISSUE TITLE
Rule to enforce not to use constant numbers for height and width

### ISSUE THREAD

**Issue by rodrigobastosv** [pattobrien/design_system_lints#4]
First of all, thanks for the work. This is a great idea.

I would love to see a rule that force me not to use constant values in properties like height and width of elements.

We often use stuff like this here https://pub.dev/packages/flutter_screenutil to make our apps responsive. That rule will enforce the use of that.

**Comment by pattobrien** [comment]
Thanks for the kind words, Rodrigo!

This is a great idea for an "adaptability" rule - I did a bit of looking into `flutter_screenutil` and created a quick demo of a solution using DSL, but please correct me if there's something I'm missing for your particular use case.

At its core, design system lints enforce UI  _consistency_. So what design system lints would recommend you do would be the following:

```dart
@designSystem
class MyDesignSystem {
  static final largeBox = 360.sp;
}

class ShopPage extends StatelessWidget {
  const ShopPage({super.key});

  @override
  Widget build(BuildContext context) {
    return SizedBox(
      width: MyDesignSystem.largeBox, // no lint appears
      child: SizedBox(
        // lint appears below for both width and height, as the numbers
        // didn't come from the design system and may look bad / inconsistent,
        // the more we use random height/width numbers throughout our app
        width: 297.h,
        height: 296.h,
        child: Text('Shop'),
      ),
    );
  }
}

```

In other words, DSL would recommend you not use `297.h` and `298.h` outside of a design system at all. This is so all of your widgets could reference the same X amount of Design System height/width values, so that developers cant just use any size number they want.

However, this would mean that your design system would need to contain all of the heights and widths that you would want to use throughout your app, and that may not be what you're looking to do.

What are your thoughts? Ultimately I think this rule is still 100% valid, but maybe I could create this under an `adaptive_lints` package instead, since the goal of DSL is technically different.

**Comment by rodrigobastosv** [comment]
Hey @pattobrien , thanks very much for the fast reply. I got your point and agree with everything you said. It's not going to be very productive to put all the heights and widths under the Design System.

I love the idea of the creation of another package just for that. Thanks very much!
